bash_automation.md

```diff
@@ -0,0 +1,165 @@
+# Automating workflows using bash
+
+## Objectives
+
++ Write a shell script with multiple variables.
++ Incorporate a for loop into a shell script.
++ Understand how to make a workflow more efficient and less error-prone
+
+
+We have now worked through two workflows, RNA-seq analysis and variant calling.
+In both cases, we pasted all of our commands into the terminal, and used for 
+loops to run each step of the workflow over our six samples. This is a great way
+to understand how each tool works and to establish a workflow you want to use.
+
+Let's imagine now that we decided to download all 672 samples from this dataset
+instead of only working with our six samples. We already know how each tool in
+our workflow works, and we know how to run them. We can now use bash scripting
+to automate our workflow. 
+
+We are going to automate the quality control portion of our workflow. To write 
+a script to run our FastQC analysis, we’ll take each of the commands we entered 
+to run FastQC and process the output files and put them into a single file with 
+a `.sh` extension. The `.sh` is not essential, but serves as a reminder to 
+ourselves and to the computer that this is a shell script.
+
+## Starting with automation
+
+We will be constructing our bash script in RStudio. We will construct our bash
+script in a text file in the "script" pane of RStudio, and we will execute it 
+in the "Terminal" tab of the "console" pane. We could use nano, but RStudio
+makes it easier to navigate a text file. 
+
+First, let's make a new directory called `scripts`.
+
+```
+cd ~
+mkdir scripts
+```
+
+## Constructing a bash script
+
+When we write a bash script, we need to add *all* commands that we ran in our
+workflow. This includes making and changing directories, moving files, and 
+running analysis programs like `fastqc`. 
+
+Let's start by changing into the right directory and running `fastqc`:
+
+```
+cd ~/data/
+
+fastqc *fastq.gz
+```
+
+That's it! We now have a bash script that automates `fastqc`! 
+
+Let's run the bash script from our terminal in RStudio
+
+```
+bash qc.sh
+```
+
+Let's add more to our script. Next we'll organize our output files. 
+
+```
+cd ~/data/
+
+fastqc *.fastq.gz
+
+mkdir -p ~/fastqc_untrimmed
+
+mv *.zip ~/fastqc_untrimmed/
+mv *.html ~/fastqc_untrimmed/
+``` 
+
+We can also add a few pointers that
+will print to our screen to let us know what point in the workflow we are in.
+
+```
+cd ~/data/
+
+echo "Running FastQC ..."
+fastqc *.fastq.gz
+
+mkdir -p ~/fastqc_untrimmed
+
+echo "Saving FastQC results..."
+mv *.zip ~/fastqc_untrimmed/
+mv *.html ~/fastqc_untrimmed/
+```
+
+Let's run this again.
```

Comment:
even if kept in R, there needs to be communication about hopping between the RScript window, and the "Terminal" tab in the "Console" window i think

Reply:
pshhhh nah people can guess! 

just kidding yes I totally agree. when we decide which way we're going, i'll add more sign posting.